client/ayon_maya/plugins/publish/validate_clashing_sibling_names.py

```diff
@@ -0,0 +1,167 @@
+from collections import defaultdict
+import inspect
+
+import ayon_maya.api.action
+from ayon_core.pipeline.publish import (
+    OptionalPyblishPluginMixin,
+    PublishValidationError,
+    ValidateContentsOrder,
+)
+from ayon_maya.api import plugin, lib
+import pyblish.api
+
+
+def remove_namespace(path: str) -> str:
+    """Remove namespace from full path.
+
+    Example:
+        >>> remove_namespace("|aa:bb:foo|aa:bb:bar|cc:hello|dd:world")
+        '|foo|bar|hello|world'
+
+    Arguments:
+        path (str): Full node path.
+
+    Returns:
+        str: Node path with namespaces removed.
+    """
+    return "|".join(
+        name.rsplit(":", 1)[-1] for name in path.split("|")
+    )
+
+
+class ValidateClashingSiblingNames(plugin.MayaInstancePlugin,
+                                   OptionalPyblishPluginMixin):
+    """Validate siblings have unique names when namespaces are stripped."""
+
+    order = ValidateContentsOrder
+    families = ["pointcache", "animation", "usd"]
+    label = "Validate clashing sibling names"
+    actions = [ayon_maya.api.action.SelectInvalidAction]
+
+    @staticmethod
+    def get_invalid(instance):
+        """Return all nodes that have non-unique names with siblings when
+        namespaces are stripped.
+
+        Returns:
+            list[str]: Non-unique siblings
+        """
+        stripped_name_to_full_path = defaultdict(set)
+        for node in instance:
+            stripped_name = remove_namespace(node)
+            stripped_name_to_full_path[stripped_name].add(node)
+
+        invalid: "list[str]" = []
+        for _stripped_name, nodes in stripped_name_to_full_path.items():
+            if len(nodes) > 1:
+                invalid.extend(nodes)
+
+        if invalid:
+            # We only care about the highest conflicts since child conflicts
+            # only occur due to the conflicts higher up anyway
+            invalid = lib.get_highest_in_hierarchy(invalid)
+
+        return invalid
+
+    def process(self, instance):
+        """Process all the nodes in the instance "objectSet"""
+        if not self.is_active(instance.data):
+            return
+
+        if not self.is_strip_namespaces_enabled(instance):
+            return
+
+        invalid = self.get_invalid(instance)
+        if invalid:
+
+            report_list = "\n".join(f"- {node}" for node in sorted(invalid))
+
+            raise PublishValidationError(
+                "With stripped namespaces there are conflicting sibling names "
+                "that are not unique:\n"
+                f"{report_list}",
+                description=self.get_description())
+
+    def is_strip_namespaces_enabled(self, instance) -> bool:
```

Comment:
@iLLiCiTiT this really seems like madness - but I'm admittedly not entirely sure how to do this better. Many of these plugins can be enabled/disabled, they can be optional and hence disabled for the instance only, they can either have exposed publish attributes for `stripNamespaces` or have it locked down and defined by settings as defaults.

Reply:
That actually sounds more legit to me. Single plugin that is capable of showing all the input fields and based on the checked fields adds families (or different data) to instance. But that is a lot of work, so I don't know if should be handled in this PR.

`OptionalPyblishPluginMixin` was not really meant that it is dependent on other plugins. BTW shouldn't the other plugin add something to instance data (for now) instead of having this logic?

Reply:
> OptionalPyblishPluginMixin was not really meant that it is dependent on other plugins. BTW shouldn't the other plugin add something to instance data (for now) instead of having this logic dependent on current state of instance (might not represent state of instance at the moment the other plugin was processed)?

The other plugins are Extractors - they run after the validators and hence the data is not there yet, so I'd need to move those to a Collector instead - which means we get into the shitshow of backwards compatibilities, etc. :)

This logic here is the best I could think of to support it now without refactoring everything.

Reply:
I do agree "correct implementation" would require massive refactor to allow it. I'll let @antirotor to decide.